projects/project-player-library/src/lib/pages/task-details-page/task-details-page.component.ts

```diff
@@ -1,20 +1,114 @@
 import { Component, OnInit } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
 import { ActivatedRoute } from '@angular/router';
+import { DailogPopupComponent } from '../../shared/dialog-popup/dailog-popup.component';
+import { EditTaskCardComponent } from '../../shared/edit-task-card/edit-task-card.component';
+import { projectDetailsData } from '../../project-details.component.spec.data';
 
 @Component({
   selector: 'lib-taskdetails-page',
-  // standalone: true,
-  // imports: [],
   templateUrl: './task-details-page.component.html',
-  styleUrl: './task-details-page.component.css'
+  styleUrl: './task-details-page.component.css',
 })
-export class TaskdetailspageComponent implements OnInit{
-  constructor(private route:ActivatedRoute){}
+export class TaskDetailsPageComponent implements OnInit {
+  constructor(private route: ActivatedRoute, private dialog: MatDialog) {}
+  taskId: any;
+  selectedValue!: string;
+  textFormControl = new FormControl('');
+  task: any = {};
+  taskOptions = [
+    {
+      viewValue: 'Not Started',
+      value: 'NOTstarted',
+    },
+    {
+      viewValue: 'In Progress',
+      value: 'inProgress',
+    },
+    {
+      viewValue: 'Completed',
+      value: 'completed',
+    },
+  ];
+
   ngOnInit(): void {
-    this.route.paramMap.subscribe((params:any)=>{
+    this.route.paramMap.subscribe((params: any) => {
       const id = params.get('id');
       console.log(id);
+      this.taskId = id;
+    });
+    this.getTaskDetails();
+  }
+  getTaskDetails() {
+    this.task = projectDetailsData.tasks.find(
+      (task) => task._id === this.taskId
+    );
+  }
+  addSubTask(data: any) {
+    console.log(data);
+    this.task.children.push({ name: data, status: 'NOTstarted' });
+    this.updateTaskStatus();
+    this.textFormControl.reset();
+  }
+  editTask() {
+    this.openEditTaskName('0', '0', this.task.name,"EDIT_TASK"
+    );
+  }
+  openEditTaskName(
+    enterAnimationDuration: string,
+    exitAnimationDuration: string,
+    taskName: string,
```

Comment:
Giving taskName and editType here, is it required??

Reply:
1. taskName this is for editing of the task name or subtask name 
2. editType is type of modal used for task or subtask